src/main/java/teammates/logic/core/EmailGenerator.java

```diff
@@ -87,6 +89,87 @@
     }
     
     /**
+     * Generates the summary of the feedback sessions (in which at least one
+     * email has been sent to students) email for the given {@code courseId} for {@code student}
+     */
+    public EmailWrapper generateFeedbackSessionResendAllLinksEmail(String courseId,
+            StudentAttributes student) {
+        
+        CourseAttributes course = coursesLogic.getCourse(courseId);
+        
+        List<FeedbackSessionAttributes> sessions = new ArrayList<FeedbackSessionAttributes>();
+        List<FeedbackSessionAttributes> fsInCourse = fsDb.getFeedbackSessionsForCourse(courseId);
+        
+        for (FeedbackSessionAttributes fsa : fsInCourse) {
+            if (!fsa.isPrivateSession() && (fsa.isSentOpenEmail() || fsa.isSentPublishedEmail())) {
+                sessions.add(fsa);
+            }
+        }
+        
+        String submitUrl = null;
+        String reportUrl = null;
+        
+        StringBuffer linksFragmentValue = new StringBuffer(1000);
+        String joinUrl = Config.getAppUrl(student.getRegistrationUrl()).toAbsoluteString();
+        
+        String joinFragmentValue = "";
```

Comment:
This can be simplified to:
```java
String joinFragmentValue = isYetToJoinCourse(student)
                           ? Templates.populateTemplate(...)
                           : "";
```